[Feat] 카테고리 수정 API

BAEK0111 opened the pull request:
## #️⃣연관된 이슈
> #35 

## 📝작업 내용
> 카테고리 수정 api 구현

## 🔎코드 설명
> 카테고리의 이름과 색상을 입력받고 수정한다.

## 💬고민사항 및 리뷰 요구사항 (Optional)
> X

## 비고 (Optional)
> 참고했던 링크 등 참고 사항을 적어주세요. 코드 리뷰하는 사람이 참고해야 하는 내용을 자유로운 형식으로 적을 수 있습니다.
<img width="795" alt="스크린샷 2025-01-18 오후 11 42 45" src="https://github.com/user-attachments/assets/cbb606ca-8a3f-4994-9246-27d303a90dea" />
<img width="811" alt="스크린샷 2025-01-19 오후 6 18 51" src="https://github.com/user-attachments/assets/392fc9a8-16a4-42d6-a3d1-1cc5307f99e4" />
<img width="786" alt="스크린샷 2025-01-20 오전 1 24 39" src="https://github.com/user-attachments/assets/fabc02d9-c6e1-4f62-9f15-b1c058c95e86" />

BAEK0111 committed:
### 제목 - (영문기준) 50자 이내
# [커밋 타입]: [#이슈번호] [작업 내용]
# ex) feat: #3 유저 삭제 기능 구현
feat: #35 카테고리 수정 api 구현

### 본문 - (영문 기준) 1줄 72자 이내
# (1)무엇을 (2)왜 (3) 어떻게 수정했는지
카테고리의 이름과 색상을 다시 입력받아 수정하는 기능 구현

### 꼬리말
# (선택)이슈 등 아래 작성

#   [ 커밋 타입 목록 ]
#   feat	:  기능 (새로운 기능)
#   fix		:  버그 (버그 수정)
#   refactor	:  리팩토링
#   style	:  스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs	:  문서 (문서 추가, 수정, 삭제)
#   test	:  테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore	:  기타 변경사항, 설정 관련 변경사항 (빌드 스크립트 수정, package.json 등)

Jinho622 commented on src/main/java/TtattaBackend/ttatta/domain/enums/CategoryColor.java:
```diff
@@ -2,4 +2,12 @@
 public enum CategoryColor {
     RED, ORANGE, YELLOW, GREEN, SKYBLUE, BLUE, INDIGO, VIOLET;
+
+    public static CategoryColor fromString(String color) {
```
소문자 처리와 에러 처리도 잘 해주셨는데 로직 처리는 모두 service에서 해주면 좋을 것 같아요. fromString을 DiaryCategoryCommandServiceImpl로 옮겨 주시면 좋을 것 같습니다!
그리고 에러 처리는 workbook 7주차랑 다른분들 코드 참고하셔서 handler사용해서 처리해주시면 감사하겠습니다:)

Jinho622 commented:
그리고 pr 제목 convention에 맞게 수정 부탁드립니다!

BAEK0111 committed:
### 제목 - (영문기준) 50자 이내
# [커밋 타입]: [#이슈번호] [작업 내용]
# ex) feat: #3 유저 삭제 기능 구현
feat: #35 카테고리 색상 추가

### 본문 - (영문 기준) 1줄 72자 이내
# (1)무엇을 (2)왜 (3) 어떻게 수정했는지
카테고리 4개의 색상 추가

### 꼬리말
# (선택)이슈 등 아래 작성

#   [ 커밋 타입 목록 ]
#   feat	:  기능 (새로운 기능)
#   fix		:  버그 (버그 수정)
#   refactor	:  리팩토링
#   style	:  스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs	:  문서 (문서 추가, 수정, 삭제)
#   test	:  테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore	:  기타 변경사항, 설정 관련 변경사항 (빌드 스크립트 수정, package.json 등)

BAEK0111 committed:
### 제목 - (영문기준) 50자 이내
# [커밋 타입]: [#이슈번호] [작업 내용]
# ex) feat: #3 유저 삭제 기능 구현
feat: #35 카테고리 색상 추가

### 본문 - (영문 기준) 1줄 72자 이내
# (1)무엇을 (2)왜 (3) 어떻게 수정했는지
카테고리 존재 여부 에러 헨들러 처리
DiaryCategoryRequestDTO 에서 카테고리 타입 수정(String -> CategoryColor)

### 꼬리말
# (선택)이슈 등 아래 작성

#   [ 커밋 타입 목록 ]
#   feat	:  기능 (새로운 기능)
#   fix		:  버그 (버그 수정)
#   refactor	:  리팩토링
#   style	:  스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs	:  문서 (문서 추가, 수정, 삭제)
#   test	:  테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore	:  기타 변경사항, 설정 관련 변경사항 (빌드 스크립트 수정, package.json 등)

BAEK0111 committed:
feat: #35 에러 핸들러 추가

카테고리 색상 존재 여부 에러 헨들러 처리

BAEK0111 committed:
fix: #35 에러 핸들러 수정

카테고리 존재 여부 커스텀 어노테이션 처리

BAEK0111 committed:
feat: #35 에러 핸들러 수정

카테고리 색상 존재 여부 커스텀 어노테이션 처리

BAEK0111 committed:
refactor: #35 에러 핸들러 수정

코드 깔끔하게 수정(주석 삭제)
색상 validation에 null 추가
